### PR TITLE
Spot-128 Make every pipeline to return results to main program

### PR DESCRIPTION
**Pull request by rabarona**
This PR implements changes requested on JIRA issue SPOT-128. Includes the elimination of any call to save/write to file system in any place other than the main application.
Also, includes some code cleanup and refactoring.

#### Main changes

- [x] Modified the run method in each pipeline so that it returns two data frames, one with the final results (scored) and another with the invalid records. 

- [x] Removed any remaining call to save so that the only part of code writing to the file system is the main application.

- [x] Modified SuspiciousConnects main application to write the results from whatever pipeline is being analyzed.

#### Other code refactoring and cleanup:

- [x] Modified filter and select methods in each pipeline (*SuspiciousConnectsAnalysis.scala) to just return a set of clean or invalid records removing the select call in case users want to do further analysis and do not drop columns.

- [x] Removed the concept of corrupt records leaving only invalid records. Updated filtering/validations so only valid records are processed.

- [x] Code refactoring/consistency on *SuspiciousConnectsModel.scala changing trainNewModel to trainModel
- [x] Fixed a couple of typos in DNSSuspiciousConnectsAnalysisTest.scala

### CODE REVIEW

File: spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala [hunk 1]
```diff
@@ -48,58 +49,38 @@ object DNSSuspiciousConnectsAnalysis {
...
+    logger.info("Fitting probabilistic model to data")
+    val model =
+      DNSSuspiciousConnectsModel.trainModel(sparkContext, sqlContext, logger, config, dnsRecords)
 
-    val filteredDNSRecords = filterScoredDNSRecords(scoredDNSRecords, config.threshold)
+    logger.info("Identifying outliers")
+    val scoredDNSRecords = model.score(sparkContext, sqlContext, dnsRecords, config.userDomain)
 
-    val orderedDNSRecords = filteredDNSRecords.orderBy(Score)
+    val filteredScored = filterScoredRecords(scoredDNSRecords, config.threshold).orderBy(Score)
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala]: there are two calls to orderBy(Score) here ... pretty sure that it gets optimized out, but still...

File: spot-ml/src/main/scala/org/apache/spot/proxy/ProxySuspiciousConnectsAnalysis.scala [hunk 2]
```diff
@@ -40,66 +41,45 @@ object ProxySuspiciousConnectsAnalysis {
...
-          inputProxyRecords: DataFrame) = {
+          inputProxyRecords: DataFrame): SuspiciousConnectsAnalysisResults = {
 
     logger.info("Starting proxy suspicious connects analysis.")
 
-    val cleanProxyRecords = filterAndSelectCleanProxyRecords(inputProxyRecords)
+    val proxyRecords = filterRecords(inputProxyRecords)
+      .select(InSchema: _*)
+      .na.fill(DefaultUserAgent, Seq(UserAgent))
+      .na.fill(DefaultResponseContentType, Seq(ResponseContentType))
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/proxy/ProxySuspiciousConnectsAnalysis.scala]: is this imputation behavior documented somewhere?

File: spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala (hunk 1 above)
Comment (rabarona) [inline spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala]: I removed that but then it got reverted when I did rebase with master. Will change again, thanks.

File: spot-ml/src/main/scala/org/apache/spot/proxy/ProxySuspiciousConnectsAnalysis.scala (hunk 2 above)
Comment (rabarona) [inline spot-ml/src/main/scala/org/apache/spot/proxy/ProxySuspiciousConnectsAnalysis.scala]: I don't think so, it has been there for a while but no documentation. I'm going to add a JIRA issue for that documentation.

**Comment by NathanSegerlind** [comment]
+1

**Comment by lujacab** [comment]
+1

**Comment by raypanduro** [comment]
+1

**Pull request merged**